### PR TITLE
[ESI][XRT] Start of MMIO: magic number and version

### PR DESCRIPTION
**Pull request by teqdruid**
Implements a very basic XRT register MMIO space. Has the ESI magic number and version number.

### CODE REVIEW

**Comment by teqdruid** [comment]
@jptech / @alandy could y'all give this a review?

File: frontends/PyCDE/src/bsp/xrt.py [hunk 1]
```diff
@@ -39,14 +89,85 @@ def XrtBSP(user_module):
...
+      for (addr, val) in rd_addr_data.items():
+        rd_space[addr] = val
+
+      # Create the address index signal and do the muxing.
+      addr_slice = self.axil_in.araddr.slice(
+          types.int(axil_addr_width)(0), max_addr_log2)
+      rd_addr = addr_slice.reg(clk, rst)
+      rvalid = self.axil_in.arvalid.reg(clk, rst)
+      rdata = types.array(types.int(axil_data_width),
+                          len(rd_space))(rd_space)[rd_addr]
```
Comment (alandy) [inline frontends/PyCDE/src/bsp/xrt.py:149]: I suggest registering the read data mux output

```suggestion
      rvalid = self.axil_in.arvalid.reg(clk, rst, cycles=2)
      rdata = types.array(types.int(axil_data_width),
                          len(rd_space))(rd_space)[rd_addr].reg(clk)
```

File: frontends/PyCDE/src/bsp/xrt.py [hunk 2]
```diff
@@ -39,14 +89,85 @@ def XrtBSP(user_module):
...
+          16: Bits(32)(MagicNumberLo),
+          20: Bits(32)(MagicNumberHi),
+          24: Bits(32)(VersionNumber),
+      }
+
+      # Create an array out of the sparse value map 'rd_addr_data' and zero
+      # constants. Then create a potentially giant mux. There's probably a much
+      # better way to do this. I suspect this is a common high-level construct
+      # which should be automatically optimized, but I'm not sure how common
+      # this actually is.
```
Comment (alandy) [inline frontends/PyCDE/src/bsp/xrt.py:132]: This seems like a straightforward and useful addition to the existing Mux construct

**Comment by teqdruid** [comment]
Thanks for the review @alandy !

**Pull request merged**